### PR TITLE
Fixing Async methods to always return Tasks.  Making them easier to unit test.

### PR DESCRIPTION
**Pull request by alfein**
Fixing issue: Buggy and difficult to test AsyncExtensions #2154

Modified TaskExtensions to always return a Task, because it's not a good practice to return a null, since clients could be awaiting the result, which assumes that a Task is always returned.

Had to expose Dispose() method directly on the callback for the Callbacks and the Visitor classes. Calling dispose if the result of the operation is a false.

Also exposed a Task that's created when attempting to set the result of the task asynchronously - need this to await in unit tests.  

Replaced Task.Factory.StartNew with its more modern equivalent (Task.Run).

### CODE REVIEW

File: CefSharp/Callback/ICompletionCallback.cs [hunk 1]
```diff
@@ -14,5 +16,10 @@ public interface ICompletionCallback : IDisposable
         /// Method that will be called once the task is complete. 
         /// </summary>
         void OnComplete();
+
+        /// <summary>
+        /// Returns a Task that was used to set the result of the CookieCallback 
+        /// </summary>
+        Task SetResultTask { get; }
```
Comment (amaitland) [inline CefSharp/Callback/ICompletionCallback.cs]: The Task had no place being part of the call back and needs to be removed from all instances

**Comment by amaitland** [comment]
Please give this PR a meaningful name

Will give a proper review in a few months, more changes will be required

File: CefSharp/Callback/ICompletionCallback.cs (hunk 1 above)
Comment (alfein) [reply CefSharp/Callback/ICompletionCallback.cs]: Then how do I wait for the result to be set?

Comment (amaitland) [reply CefSharp/Callback/ICompletionCallback.cs]: How is that relevant to the interfaces?

Comment (alfein) [reply CefSharp/Callback/ICompletionCallback.cs]: Removed.  Also pull in your change.

**Comment by amaitland** [comment]
Closing this as I don't believe it's necessary to expose the inner Task for testing purposes. See https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Test/Framework/AsyncExtensionFacts.cs#L37 for an example of how a callback can be tested, mimics the real world usage scenario where Continue and Dispose are always called on one of the CEF threads. If you are unhappy with this you are totally free to implement the callback interface in your code anyway you like. By all means create a new PR that adds more test cases. 

The underlying bug where the extensions incorrectly returned null was fixed a while ago.

**Pull request closed without merging**